src/migrations/upgrade_check.rs

```diff
@@ -197,7 +197,7 @@ async fn do_check(
         .with_fs()
         .build()
         .context("cannot build connection params")?;
-    let cli = &mut Connection::connect(&config, QUERY_TAG).await?;
+    let cli = &mut Box::pin(Connection::connect(&config, QUERY_TAG)).await?;
```

Comment:
Rather than putting these in the caller, let's change the functions themselves to by non-async and return `impl Future + Unpin`. The changes will be less invasive.

Reply:
Do you mean this specific call or all of the new `Box::pin`s?

Reply:
I see, you mean just `Connection::connect`, because it is used in a lot of places.

Reply:
I think `Connection::connect` would be a good candidate. Looking through the others, I don't really see any other commonly-used and large futures.